### PR TITLE
[IMP] stock: auto-printing and report cleanups

### PR DESCRIPTION
**Pull request by ticodoo**
This PR handles a number of stock/mrp related printing improvements:
- allow users to select certain reports to be auto-printed at validation of a picking or a manufacturing order as well as when "Put in Pack" is used in a picking and the "Create Serial Number" button is clicked within an mo (in differing use cases). Each report was added in separate commits in case individual reports are decided to no longer be allowed to auto-print in the future (and also avoid a giant commit for all the reports at the same time)
- product label templates have been split into separate reports to allow for easier user customization (i.e. via studio)

It also handles some other related-ish improvements/fixes:
- unify shipping doc names: before they were inconsistent across shipping connectors, so now there is a method that should be called by shipping connectors to determine their pdf filename prefixes
- reception report correctly split moves when assigning (to prevent incorrect `reservation_dates` in moves)
- avoid asking for document layout configuration when printing product labels (it doesn't use the layout template)
- avoid singleton access when validating 2 MOs with different operation types

Additional detail is in each commit
Task: 3046178
ENT PR: odoo/enterprise#43362
Upgrade PR: odoo/upgrade#5298

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by clesgow — COMMENTED** [review]
Only got a few nitpicks left over what have already been discussed

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -1513,61 +1520,69 @@ def _check_destinations(self, move_line_ids):
...
+        move_lines_to_pack.write({
+            'result_package_id': package.id,
+        })
+        if create_package_level:
+            self.env['stock.package_level'].create({
+                'package_id': package.id,
+                'picking_id': self.id,
+                'location_id': False,
+                'location_dest_id': move_lines_to_pack.mapped('location_dest_id').id,
+                'move_line_ids': [(6, 0, move_lines_to_pack.ids)],
```
[the remark this answers is not in the document]
Comment (ticodoo) [reply addons/stock/models/stock_picking.py:1603]: :sloth: (I fixed it)

File: addons/stock/models/stock_picking.py [hunk 2]
```diff
@@ -1513,61 +1520,69 @@ def _check_destinations(self, move_line_ids):
...
+        if package_id and self.picking_type_id.auto_print_package_label:
+            if self.picking_type_id.package_label_to_print == 'pdf':
+                action = self.env.ref("stock.action_report_quant_package_barcode_small").report_action(package_id.id, config=False)
+            elif self.picking_type_id.package_label_to_print == 'zpl':
+                action = self.env.ref("stock.label_package_template").report_action(package_id.id, config=False)
+            if action:
+                action.update({'close_on_report_download': True})
+                clean_action(action, self.env)
+                return action
+        return package_id
```
[the remark this answers is not in the document]
Comment (ticodoo) [reply addons/stock/models/stock_picking.py:1618]: Agreed, but that's what the former behavior was (or it returned nothing depending on the flow) and as far as I can tell that `package_id` is never used. I think it was there simply to have the action return something (I think I read somewhere that the default used to be to return `True` when in doubt, but I don't think it's necessary anymore? :man_shrugging: ) I.e. this ~~junk~~ code probably needs to be cleaned up

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**